velox/exec/tests/PlanBuilderTest.cpp

```diff
@@ -250,4 +253,50 @@ TEST_F(PlanBuilderTest, missingOutputType) {
       "outputType must be specified");
 }
 
+TEST_F(PlanBuilderTest, projectExpressions) {
+  // Non-typed Expression
+  auto data = ROW({"c0"}, {BIGINT()});
+  VELOX_CHECK_EQ(
+      PlanBuilder()
+          .tableScan("tmp", data)
+          .projectExpressions(
+              {std::make_shared<core::FieldAccessExpr>("c0", std::nullopt)})
+          .planNode()
+          ->toString(true, false),
+      "-- Project[1][expressions: (c0:BIGINT, ROW[\"c0\"])] -> c0:BIGINT\n"); // simple field access
+  data = ROW({"c0"}, {ROW({"field0"}, {BIGINT()})});
+  VELOX_CHECK_EQ(
+      PlanBuilder()
+          .tableScan("tmp", data)
+          .projectExpressions({std::make_shared<core::FieldAccessExpr>(
+              "field0",
+              std::nullopt,
+              std::vector<core::ExprPtr>{
+                  std::make_shared<core::FieldAccessExpr>(
+                      "c0", std::nullopt)})})
+          .planNode()
+          ->toString(true, false),
+      "-- Project[1][expressions: (field0:BIGINT, ROW[\"c0\"][field0])] -> field0:BIGINT\n"); // dereference test using field access query
+
+  // Test Typed Expressions
+  VELOX_CHECK_EQ(
+      PlanBuilder()
+          .tableScan("tmp", ROW({"c0"}, {ROW({BIGINT()})}))
+          .projectExpressions(
+              {std::make_shared<core::FieldAccessTypedExpr>(BIGINT(), "c0")})
+          .planNode()
+          ->toString(true, false),
+      "-- Project[1][expressions: (p0:BIGINT, \"c0\")] -> p0:BIGINT\n");
+  VELOX_CHECK_EQ(
+      PlanBuilder()
+          .tableScan("tmp", ROW({"c0"}, {ROW({BIGINT()})}))
```

Comment:
nit: Impossible to dereference a varchar field from a ROW({BIGINT()}). Would be better to change it to ROW({VARCHAR()}).